client/cache_test.go

```diff
@@ -0,0 +1,145 @@
+// Copyright Contributors to the Open Cluster Management project
+package client
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+)
+
+var _ = Describe("Test the cache", Ordered, func() {
+	var cache ObjectCache
+	configMapGVK := schema.GroupVersionKind{
+		Version: "v1",
+		Kind:    "ConfigMap",
+	}
+
+	BeforeEach(func() {
+		discoveryClient, err := discovery.NewDiscoveryClientForConfig(k8sConfig)
+		Expect(err).ToNot(HaveOccurred())
+
+		cache = NewObjectCache(discoveryClient)
+	})
+
+	It("It returns an error for queries with no cache entries", func() {
+		object, err := cache.Get(configMapGVK, "default", "something")
+		Expect(err).To(MatchError(ErrNoCacheEntry))
+		Expect(object).To(BeNil())
+
+		objects, err := cache.List(configMapGVK, "default", nil)
+		Expect(err).To(MatchError(ErrNoCacheEntry))
+		Expect(objects).To(BeNil())
+	})
+
+	It("It caches a get object", func() {
+		object := unstructured.Unstructured{}
+		object.SetName("cached-object1")
+		cache.CacheObject(configMapGVK, "default", "something", &object)
```

Comment:
I am confused here. I thought `Get`  method's name arg is a resource name. but here, names are diff
```
	objID := ObjectIdentifier{
		Group:     gvk.Group,
		Version:   gvk.Version,
		Kind:      gvk.Kind,
		Namespace: namespace,
		Name:      name,
	}
```

Reply:
Also `List`  have multiple objects under the same name? Could you explain it to me?

Reply:
Do you still have this question after our meeting?